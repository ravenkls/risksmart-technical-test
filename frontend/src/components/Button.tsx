import React, { ComponentProps } from "react";

type Props = {
  children?: React.ReactNode;
  ghost?: boolean;
  danger?: boolean;
  onClick?: ComponentProps<"button">["onClick"];
  className?: string;
};

function Button({ children, ghost, onClick, className, danger }: Props) {
  return (
    <button
      className={`p-2 ${
        ghost
          ? "bg-white text-gray-900 ring-1 ring-inset ring-gray-300 hover:bg-gray-50"
          : "text-white bg-primary-600 hover:bg-primary-500"
      } inline-flex w-full sm:w-auto justify-center rounded-md shadow-sm px-3 py-2 text-sm font-semibold ${
        danger ? "text-red-600" : ""
      } ${className ?? ""}`}
      onClick={onClick}
    >
      {children}
    </button>
  );
}

export default Button;
