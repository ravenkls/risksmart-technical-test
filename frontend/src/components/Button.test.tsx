import { render, fireEvent } from "../../test/test-utils";
import Button from "./Button";

describe("Button", () => {
  it("renders children", () => {
    const { getByText } = render(<Button>Click me</Button>);
    expect(getByText("Click me")).toBeInTheDocument();
  });

  it("matches snapshot", () => {
    const { getByText } = render(<Button>Click me</Button>);
    expect(getByText("Click me")).toMatchSnapshot();
  });

  it("calls onClick when clicked", () => {
    const handleClick = jest.fn();
    const { getByText } = render(
      <Button onClick={handleClick}>Click me</Button>
    );
    fireEvent.click(getByText("Click me"));
    expect(handleClick).toHaveBeenCalled();
  });

  it("applies danger styles when danger prop is true", () => {
    const { getByText } = render(<Button danger>Click me</Button>);
    expect(getByText("Click me")).toHaveClass("text-red-600");
    expect(getByText("Click me")).toMatchSnapshot();
  });

  it("applies ghost styles when ghost prop is true", () => {
    const { getByText } = render(<Button ghost>Click me</Button>);
    expect(getByText("Click me")).toHaveClass("bg-white");
    expect(getByText("Click me")).toMatchSnapshot();
  });

  it("applies ghost and danger styles when ghost prop is true", () => {
    const { getByText } = render(
      <Button ghost danger>
        Click me
      </Button>
    );
    expect(getByText("Click me")).toMatchSnapshot();
  });
});
