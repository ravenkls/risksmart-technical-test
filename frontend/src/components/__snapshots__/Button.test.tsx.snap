// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button applies danger styles when danger prop is true 1`] = `
<button
  class="p-2 text-white bg-primary-600 hover:bg-primary-500 inline-flex w-full sm:w-auto justify-center rounded-md shadow-sm px-3 py-2 text-sm font-semibold text-red-600 "
>
  Click me
</button>
`;

exports[`Button applies ghost and danger styles when ghost prop is true 1`] = `
<button
  class="p-2 bg-white text-gray-900 ring-1 ring-inset ring-gray-300 hover:bg-gray-50 inline-flex w-full sm:w-auto justify-center rounded-md shadow-sm px-3 py-2 text-sm font-semibold text-red-600 "
>
  Click me
</button>
`;

exports[`Button applies ghost styles when ghost prop is true 1`] = `
<button
  class="p-2 bg-white text-gray-900 ring-1 ring-inset ring-gray-300 hover:bg-gray-50 inline-flex w-full sm:w-auto justify-center rounded-md shadow-sm px-3 py-2 text-sm font-semibold  "
>
  Click me
</button>
`;

exports[`Button matches snapshot 1`] = `
<button
  class="p-2 text-white bg-primary-600 hover:bg-primary-500 inline-flex w-full sm:w-auto justify-center rounded-md shadow-sm px-3 py-2 text-sm font-semibold  "
>
  Click me
</button>
`;
